datadog_checks_dev/datadog_checks/dev/plugin/hatch/environment_collector.py

```diff
@@ -66,6 +66,12 @@ def format_base_package(features, version='', local=False):
     def pip_install_command(*args):
         return f'python -m pip install --disable-pip-version-check {{verbosity:flag:-1}} {" ".join(args)}'
 
+    def finalize_environments(self, config):
+        for env_config in config.values():
+            if env_config.get('python') == '2.7':
+                env_config['dev-mode'] = False
+        return config
+
```

Comment:
Could you elaborate why this is needed?

Reply:
Do you mean add a comment explaining why we don't install Python2 in editable mode anymore?

Reply:
not sure we need a comment in fact, I just don't understand how this is related to the setuptools modifications 😅 that's probably a dumb question though

Reply:
To me it's an indication that we need a comment.

So, for py3 we don't need setuptools at all. For py2 we were using setuptools to install the integrations in editable mode for testing. But nobody actually develops the py2 integrations in practice, so we're removing the editable (dev-mode in hatch-speak) installs and thus the need for setuptools.

Reply:
got it thanks!

Reply:
[PEP 660](https://peps.python.org/pep-0660/) support was added to `pip` in version [21.3](https://pip.pypa.io/en/stable/news/#v21-3) but that was after Python 2 was dropped

Reply:
thanks @ofek!